lib/Site.js

```diff
@@ -356,22 +357,55 @@ Site.prototype.generatePages = function () {
   // Render the final rendered page to the output folder.
   const { baseUrl } = this.siteConfig;
   const builtFiles = {};
-  const pages = this.siteConfig.pages || [];
   const processingFiles = [];
-  this.pageModels = pages.map(page => this.createPageData({
-    baseUrl,
-    pageSrc: page.src,
-    title: page.title,
-    pageTemplate: this.pageTemplate,
-  }));
-  this.pageModels.forEach((page) => {
-    processingFiles.push(page.generate(builtFiles));
-  });
-  return new Promise((resolve, reject) => {
-    Promise.all(processingFiles)
-      .then(resolve)
-      .catch(reject);
+  return this.buildSiteData().then((frontMatterData) => {
+    const pages = frontMatterData || [];
+    this.pageModels = pages.map(page => this.createPageData({
+      baseUrl,
+      pageSrc: page.src,
+      title: page.title,
+      pageTemplate: this.pageTemplate,
+    }));
+    this.pageModels.forEach((page) => {
+      processingFiles.push(page.generate(builtFiles));
+    });
+    return new Promise((resolve, reject) => {
+      Promise.all(processingFiles)
+        .then(resolve)
+        .catch(reject);
+    });
+  },
+  );
+};
+
+/**
+ * Builds the siteData file that specifies addressable pages and their associated keywords
+ */
+Site.prototype.buildSiteData = function () {
+  const filePaths = walkSync(this.rootPath, { directories: false })
```

Comment:
Instead of filtering, I think you can use something like `{ directories: false, ignore: [ '_site' ] }` and the `globs` property to only match `.md` files

Reply:
I tried using globs to ignore but I didn't get too far with it. I'll give your suggestion a shot.

Reply:
Actually, maybe it is not needed. `_site` folder should have been emptied

Reply:
Yes you're right, I just gave it another try. The folder is emptied at this point so the filtering for `_site` files is extraneous. I'll remove it.